src/Cardano/Transaction/Balancer/Balancer.purs

```diff
@@ -180,6 +191,9 @@ runBalancer unbalancedTx ctx = do
     maybe (liftAff ctx.walletInterface.getChangeAddress) pure
       (unwrap ctx.balancerConstraints).changeAddress
   let txWithNetwork = setTxNetwork ctx.network unbalancedTx
+  allUtxos <- getAllUtxos
+  availableUtxos <- liftAff $ ctx.walletInterface.filterLockedUtxos allUtxos
+  let spendableUtxos = getSpendableUtxos availableUtxos
```

Comment:
Why a UTXO with inline datum isn't spendable?

Reply:
If a transaction uses PlutusV1 scripts, we have to filter out all utxos that use PlutusV2 features like inline datums or reference scripts, or otherwise the transaction as a whole will be rejected